src/parser/mod.rs

```diff
@@ -11096,6 +11096,19 @@ impl<'a> Parser<'a> {
                 arg,
                 operator: FunctionArgOperator::Assignment,
             })
+        } else if dialect_of!(self is MsSqlDialect) {
+            // FUNC(<expr> : <expr>)
+            let name = self.parse_wildcard_expr()?;
+            if self.consume_token(&Token::Colon) {
+                let arg = self.parse_expr()?.into();
+                Ok(FunctionArg::Named {
+                    name,
+                    arg,
+                    operator: FunctionArgOperator::Colon,
+                })
```

Comment:
hmm yeah regarding the requirement I wondered a bit of pros/cons but I figure it might be the lesser evil to add a new variant like `FunctionArg::Association { target: Expr, arg, operator }` variant that is able to represent an arbitrary expression? 

The immediate advantage would be that its not a breaking change but we would also avoid hiding the identifier which is what users expect from a named function argument since that is a feature used in a lot of scenarios. So I figure we could be more explicit to represent this as a different class of functions, and then highlight in the doc the difference between the two

Reply:
It might be better to add a new variant, but the naming is a headache.

Reply:
How about this?

```rs
pub enum FunctionArg {
    Named {
        name: Ident,
        arg: FunctionArgExpr,
        operator: FunctionArgOperator,
    },
    ExprNamed {
        name: Expr, // <<
        arg: FunctionArgExpr,
        operator: FunctionArgOperator,
    },
    Unnamed(FunctionArgExpr),
}

pub trait Dialect {
  /// Some dialect supports `function(<arbitrary-expr>:<arbitrary-expr>,...)`,
  /// for this scenario, when returning `true` use the `FunctionArg::ExprNamed` variant in all cases to express named arguments,otherwise use the `FunctionArg::Named` variant (compatible reason).
  fn supports_function_expr_named { false }
}
```